okhttp-android/src/main/kotlin/okhttp3/android/NetworkPin.kt

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (C) 2024 Block, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.android
+
+import android.net.Network
+import okhttp3.ExperimentalOkHttpApi
+
+@ExperimentalOkHttpApi
+data class NetworkPin(val network: Network?)
```

Comment:
That a request is pinned to a network is meaningful for stats, retries.

There is an alternative where we don't recreate the OkHttp client, but the Android Dns and SocketFactory are aware of this tag.

I think I'd like to move that way.

Reply:
This is currently unused as far as any behavior is concerned. I don’t think we need this!

(If users want to keep stats on networks, they can do that with EventListeners and Interceptors that are also scoped to the OkHttpClient object.)

Reply:
Removed. I'll build a separate case, but this matches Cronet/HttpEngine functionality. 

Although not on a Request basis, which was where this was heading.